dashboard/src/components/settings/environmentVariables/SystemEnvironmentVariableSettings/SystemEnvironmentVariableSettings.tsx

```diff
@@ -21,18 +22,43 @@ import generateAppServiceUrl, {
 } from '@/utils/common/generateAppServiceUrl';
 import { LOCAL_HASURA_URL } from '@/utils/env';
 import { generateRemoteAppUrl } from '@/utils/helpers';
-import { useGetAppInjectedVariablesQuery } from '@/utils/__generated__/graphql';
+import type { JwtSecretFragment } from '@/utils/__generated__/graphql';
+import { useGetEnvironmentVariablesQuery } from '@/utils/__generated__/graphql';
 import { Fragment, useState } from 'react';
 
+function getJwtSecretsWithoutFalsyValues(jwtSecrets: JwtSecretFragment[]) {
+  return jwtSecrets.map((secret) =>
+    Object.keys(secret).reduce(
+      (secretWithoutFalsyValues, key) =>
+        secret[key] && key !== '__typename'
+          ? { ...secretWithoutFalsyValues, [key]: secret[key] }
+          : secretWithoutFalsyValues,
+      {},
+    ),
+  );
+}
+
 export default function SystemEnvironmentVariableSettings() {
   const [showAdminSecret, setShowAdminSecret] = useState(false);
   const [showWebhookSecret, setShowWebhookSecret] = useState(false);
 
   const { openDialog } = useDialog();
+  const { maintenanceActive } = useUI();
   const { currentApplication } = useCurrentWorkspaceAndApplication();
-  const { data, loading, error } = useGetAppInjectedVariablesQuery({
-    variables: { id: currentApplication?.id },
+  const { data, loading, error } = useGetEnvironmentVariablesQuery({
+    variables: { appId: currentApplication?.id },
+    fetchPolicy: 'cache-only',
   });
+
+  const { jwtSecrets, webhookSecret, adminSecret } = data?.config?.hasura || {};
```

Comment:
Should we check that `data.config.hasura` is actually set? Instead of using lots of `?` everywhere? I've seen the use of `?` in other places in relation to `data.config.*`.

What will happen if `data.config.hasura` is empty? Will the UI work?

Reply:
Then these values are just simply going to be undefined. The optional chaining is actually the mechanism that makes the UI error tolerant.